bitbucket/bitbucket.py

```diff
@@ -108,25 +116,38 @@ def set_deployment_environment_variable(self, name, workspace, uuid, key, value,
                 "secured": is_secure
                 }
 
-        r = requests.post(url=url, json=body, auth=self.auth)
+        r = requests.post(url=url, json=body, auth=self.auth, proxies=self.proxies)
         if not r.ok:
             print("Failed to set deployment environment variable")
             print(r.text)
 
-    def create_and_configure_repository(self, name, project = None, workspace = None):
-        project = project if project is not None else self.config['default_project']
-        workspace = workspace if workspace is not None else self.config['default_workspace']
-        print(f"Creating new repository {workspace}/{name} in project {project}")
-
-        if self.get_repository(name, workspace).status_code != 404:
-            print(f"Rrepository {workspace}/{name} already exists.")
-            return
-
-        self.create_repository(name, project, workspace)
-        self.enable_pipelines(name, workspace)
-        self.create_empty_commit(name, workspace, 'main', 'Initial commit')
-        for branch in self.config['default_restricted_branches']:
-            self.create_branch(name, workspace, branch)
-            for permission in self.config['default_branch_permissions']:
-                self.set_repository_branch_permission(name, workspace, branch, permission)
-        self.set_repository_branch_model(name, workspace, self.config['default_branching_model'])
+    def create_report(self, title, details, report_type, report_id, reporter, result, link, data, bitbucket_workspace, bitbucket_repo_slug, bitbucket_commit):
+        url=f"http://api.bitbucket.org/2.0/repositories/{bitbucket_workspace}/{bitbucket_repo_slug}/commit/{bitbucket_commit}/reports/{reporter}-{report_id}"
```

Comment:
Might be worth moving these API urls to constants if we need to use them anymore.

Reply:
Will do thanks.